lib/project_types/theme/commands/pull.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+module Theme
+  module Commands
+    class Pull < ShopifyCli::Command
+      options do |parser, flags|
+        parser.on('--store=STORE') { |url| flags[:store] = url }
+        parser.on('--password=PASSWORD') { |p| flags[:password] = p }
+        parser.on('--themeid=THEME_ID') { |id| flags[:themeid] = id }
+      end
+
+      def call(args, _name)
+        form = Forms::Pull.ask(@ctx, args, options.flags)
+        return @ctx.puts(self.class.help) if form.nil?
+
+        build(form.store, form.password, form.themeid, form.name)
+        ShopifyCli::Project.write(@ctx,
+                                  project_type: 'theme',
+                                  organization_id: nil)
+
+        @ctx.done(@ctx.message('theme.pull.pulled', form.name, form.store, @ctx.root))
+      end
+
+      def self.help
+        ShopifyCli::Context.message('theme.pull.help', ShopifyCli::TOOL_NAME, ShopifyCli::TOOL_NAME)
+      end
+
+      private
+
+      def build(store, password, themeid, name)
+        CLI::UI::Frame.open(@ctx.message('theme.checking_themekit')) do
+          Themekit.ensure_themekit_installed(@ctx)
+        end
+
+        @ctx.mkdir_p(name)
```

Comment:
If a directory with the same name as the theme already exists, will it be an issue if the CLI just downloads the theme files into the existing directory?

Reply:
Testing it by hand, there doesn't seem to be an issue, but it wouldn't be a good idea since that would mean potentially having a theme inside a theme. Perhaps they can all be prefixed by the current date and time?

We could also do away with making the file for the user entirely, though this would make it different from the `create` command.

Reply:
1. Good point about failing if we are inside a project; this should only work if we are not inside a theme, as it makes a new copy of a theme.
2. Yes, it's similar to `connect`; however, themes use private apps, which are different from how Rails/Node apps work. In particular, connecting a Theme to CLI means pulling from a remote source, rather than connecting a local project to Shopify, like for apps.
3. It checks and raises an error if it is a duplicate directory before checking if Theme Kit exists, I just forgot to close this comment after adding that check :D

Reply:
Sounds good to me on all counts! 🔥